src/lib/device-types/index.ts

```diff
@@ -342,26 +342,70 @@ function getDeviceTypes(): Promise<Dictionary<DeviceTypeInfo>> {
 	return fetchDeviceTypesAndReschedule();
 }
 
+function addDeviceTypeAliases(slugs: string[]): Promise<string[]> {
+	const slugSet = new Set(slugs);
+	return getDeviceTypes().then(deviceTypes => {
+		return _.reduce(
+			deviceTypes,
+			(result, v, key) => {
+				if (slugSet.has(key)) {
+					return result;
+				}
+				if (
+					v.latest &&
+					v.latest.deviceType &&
+					slugSet.has(v.latest.deviceType.slug)
+				) {
+					return _.uniq(_.concat(result, key));
+				}
+				return result;
+			},
+			slugs,
+		);
+	});
+}
+
 export const getAccessibleSlugs = (
 	api: PinejsClient,
 	slugs?: string[],
 ): Promise<string[]> => {
 	const options: AnyObject = {
 		$select: ['slug'],
 	};
-	if (slugs) {
-		options['$filter'] = {
-			slug: { $in: slugs },
-		};
-	}
-	return api
-		.get({
-			resource: 'device_type_table',
-			options,
+	return (
+		Promise.try(() => {
+			if (slugs) {
+				// We need to translate incomming slug aliases to non alias slugs to
+				// search the database
+				const slugSet = new Set(slugs);
+				return getDeviceTypes()
+					.then(deviceTypesInfos => {
+						return _(deviceTypesInfos)
```

Comment:
I think this can be simpler:
```ts
const nonAliasSlugs = slugs.reduce((acc, s) => {
  const deviceTypesInfo = deviceTypesInfos[s];
  if (deviceTypesInfo) {
    acc.push(deviceTypesInfo.latest.deviceType.slug)
  }
  return acc;
}, []);
```